### PR TITLE
first pass at client and server doxygen API

### PR DESCRIPTION
**Pull request by bigbrett**
- Add doxygen documentation for public client+server API
- clang-format

Right now it the doxygen is just present in the header. I can also put it in the `.c` file (my typical method) or can duplicate the header into a "documentation" version like wolfSSL does.

### CODE REVIEW

**Review by billphipps — DISMISSED** [review]
Looks good!  Minor nits/suggestions/ideas.  Lemme know if you want to merge as is.  I'm good with it.

File: wolfhsm/wh_client.h [hunk 1]
```diff
@@ -72,194 +72,1391 @@ typedef struct whClientConfig_t whClientConfig;
...
+ * @param[out] out_rcv_len Pointer to store the size of the received data
+ * payload.
+ * @param[out] rcv_data Pointer to store the received data payload.
+ * @return int Returns 0 on success, or a negative error code on failure.
+ */
 int wh_Client_Echo(whClientContext* c, uint16_t snd_len, const void* snd_data,
-        uint16_t *out_rcv_len, void* rcv_data);
+                   uint16_t* out_rcv_len, void* rcv_data);
 
 /** Key functions */
```
[the remark this answers is not in the document]
Comment (bigbrett) [reply wolfhsm/wh_client.h:259]: Figured that would be something the manual describes, not necessarily in the header. But probs a good idea to have in the code anyway. Added.

File: wolfhsm/wh_client.h [hunk 2]
```diff
@@ -72,194 +72,1391 @@ typedef struct whClientConfig_t whClientConfig;
...
-/* Initialize client context and connect to server based on provided
- * configuration */
+
+/**
+ * Initializes a whClientContext object with the provided configuration.
+ *
+ * @param c The pointer to the whClientContext object to be initialized.
+ * @param config The pointer to the whClientConfig object containing the
+ * configuration settings.
+ * @return Returns 0 on success, or a negative value indicating an error.
```
[the remark this answers is not in the document]
Comment (bigbrett) [reply wolfhsm/wh_client.h:82]: I think that would require every single doxygen comment to contain that reference. This is documented in the "client library" section of the manual, so think that is hopefully enough info.

File: wolfhsm/wh_server.h [hunk 3]
```diff
@@ -164,35 +164,36 @@ typedef struct {
...
 #ifdef WOLFHSM_SHE_EXTENSION
     she_context* she;
 #endif
-#if defined WOLF_CRYPTO_CB /* TODO: should we be relying on wolfSSL defines? */
+#if defined WOLF_CRYPTO_CB /* TODO: should we be relying on wolfSSL defines? \
+                            */
     int devId;
 #endif
-#endif  /* WOLFHSM_NO_CRYPTO */
+#endif /* WOLFHSM_NO_CRYPTO */
```
[the remark this answers is not in the document]
Comment (bigbrett) [reply wolfhsm/wh_server.h:178]: sure, food for a future PR

File: wolfhsm/wh_client.h [hunk 4]
```diff
@@ -72,194 +72,1391 @@ typedef struct whClientConfig_t whClientConfig;
...
+ * On the server side, this key ID is used to reference the key stored in the
+ * HSM
+ *
+ * @param[in] aes Pointer to the AES key structure.
+ * @param[in] keyId Key ID to be associated with the AES key.
+ */
 void wh_Client_SetKeyAes(Aes* aes, whNvmId keyId);
 #endif
 
 /** NVM functions */
```
[the remark this answers is not in the document]
Comment (bigbrett) [reply wolfhsm/wh_client.h:548]: IMO that would go in the manual section, not in the doxygen which is simple usage info and argument details

File: wolfhsm/wh_client.h [hunk 5]
```diff
@@ -72,194 +72,1391 @@ typedef struct whClientConfig_t whClientConfig;
...
 
 
 /** Comm component functions */
+
+/**
+ * @brief Sends a communication initialization request to the server.
+ *
+ * This function prepares and sends a communication initialization request
+ * message to the server. It populates the message with the client's ID
+ * and sends it using the communication context.
```
[the remark this answers is not in the document]
Comment (bigbrett) [reply wolfhsm/wh_client.h:138]: done

File: wolfhsm/wh_client.h [hunk 6]
```diff
@@ -72,194 +72,1391 @@ typedef struct whClientConfig_t whClientConfig;
...
+
+
+/**
+ * Sends a request to the server using the specified client context.
+ *
+ * @param c The client context.
+ * @param group The group identifier.
+ * @param action The action identifier.
+ * @param data_size The size of the data to be sent.
+ * @param data A pointer to the data to be sent.
```
[the remark this answers is not in the document]
Comment (bigbrett) [reply wolfhsm/wh_client.h:110]: done

**Comment by bigbrett** [comment]
@billphipps addressed review comments, ready to merge

**Review by billphipps — APPROVED** [review]

**Pull request merged**